core/AtlasState.cpp

```diff
@@ -48,12 +49,14 @@ namespace atlas
             supported_isa_string_, isa_file_path_ + std::string("/riscv_isa_spec.json"),
             isa_file_path_)),
         stop_sim_on_wfi_(p->stop_sim_on_wfi),
+        stf_enable_(p->stf_enable),
         hypervisor_enabled_(extension_manager_.isEnabled("h")),
         vector_state_ptr_(new VectorState()),
         inst_logger_(core_tn, "inst", "Atlas Instruction Logger"),
         finish_action_group_("finish_inst"),
         stop_sim_action_group_("stop_sim")
     {
+        stf_logger_.initialize(stf_enable_, xlen_, pc_);
```

Comment:
Instead of an `initialize` method, why not create a STF constructor that takes these arguments?